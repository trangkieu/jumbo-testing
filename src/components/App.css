
body {
  background-color: black;
  font-family: Arial, Helvetica, sans-serif;
  color: white;
}


.movie-title {
  background-color: black;
  color: white;
}
.xlarge-font {
  font-size: x-large;
}


div#popular-movies {
  font-size: 30px;
}

.green-deco1 {
  width: 10%;
  height: 5px;
  background-color: #34eb4f;
  border: none;
  border-radius: 4px;
  -ms-transform: rotate(-45deg); /* IE 9 */
  transform: rotate(-45deg); /* Standard syntax */
  padding: 0px 50px 0px 0px;
  font-size: 0;
  position: fixed;
  z-index: 2;
}

.green-deco2 {
  width: 10%;
  height: 5px;
  background-color: #e51c23;
  border: none;
  border-radius: 4px;
  -ms-transform: rotate(-45deg); /* IE 9 */
  transform: rotate(-45deg); /* Standard syntax */
  padding: 0px 300px 0px 0px;
  font-size: 0;
  position: fixed;
  z-index: 1;
}

.container {
  /*padding: 20px;*/
  min-width: 600px;
  width: 90%;
  margin: auto;
}

.logo {
  width: 100%;
  margin-left: 50%;
  padding-left: 45%;

}
.float-right {
  float: right;
}

.float-left {
  float:left;
}

.logo-image {
  width:100px;
  height:100px;
  z-index: 9999;
  /*padding-left: 45%;*/

}

input {
  /*width: 100px;*/
  /*height: 40px;*/
  /*font-family: Arial;*/
  font-size: 20px;
  border: none;
  padding-left: 10px;
  padding-top: 10px;
  padding-bottom: 10px;
  border-radius: 40px;
  z-index: 3;
}

input[type=text] {
  background-color: white;
  color: #34eb4f;
  border-radius: 40px;
  padding-left: 20px;

}

::placeholder {
  color: #34eb4f;
}

.search-input {
  /*margin-left: 30%;*/
  /*margin-right: 30%;*/
  /*display: inline;*/
  /*padding-left: 20px;*/
  /*box-sizing: border-box;*/



  width: 300px;
  height: 50px;
  /*display: flex;*/
  /*flex: 1 1 200px;*/
  /*flex-wrap: wrap;*/
  /*border: 4px solid white;*/
  /*flex-direction: row;*/


}

.inline {
  display: inline;
  /*border: 1px solid red;*/
  margin:0px;
}

.layer-10 {
  position: absolute;
  z-index: 10;
}

.layer-20 {
  position: absolute;
  z-index: 20;
}

.layer-30 {
  position: absolute;
  z-index: 30;
}
.bottom {
  padding-top: 70%;
}

.App {
  display: flex;
  flex-direction: row;
  align-items: center;
}

.errorMessage {
  color: red;
  display: inline;
  padding-left: 20px;
  box-sizing: border-box;
}


.details {
  max-width: 600px;
  margin: 0 auto;
}
.details-header {
  width: auto;
  height: 400px
}


.poster-details img {
  border: none;
  box-shadow: 0 0 20px black;
  border-radius: 20px;
  width: 200px; /* one third of backdrop-details img*/
  height: auto;
  position: absolute;
  z-index: 2;
  margin-left: 20px;
  margin-top: 200px;

}

.backdrop-details img {
  border: none;
  width: 600px;
  height: auto;
  position: absolute;
  z-index: 1;
}

.details-section {
  position: absolute;
  z-index: 3;
  float: right;
  padding-left: 250px;
  padding-top: 350px;
  height: auto;
  min-width: 600px;
}


.overview-title {
  font-size: large;
  font-weight: bold;
}

.large-font {
  font-size: large;
}

.overview {
  position: absolute;
  z-index: 4;
  padding-top: 100px;
  flex-wrap: wrap;
  margin: 20px;
  max-width: 600px;
}
.overview-desc {
  font-size: medium;
}





.xxlarge-font {
  font-size: xx-large;
}

.listing img {
  border: none;
  border-radius: 20px;
  max-width: 100%;
  max-height: 100%;

}

.listing-img-wrapper {
  padding-left: 10px;
  padding-right: 10px;
}

.title-heading {
  font-size: x-large;
  margin:10px;
}

@media (max-width: 600px) {
  .listing {
    /*max-width: 600px;*/
    /*min-width: 100px;*/
    width: 600px;
    display: flex;
    /*flex: 1 1 200px;*/
    flex-wrap: wrap;
    /*border: 4px solid white;*/
    flex-direction: row;
    /*flex:1 0 250px;*/
  }
  section > div {
    /*text-align: center;*/
    flex:0 0 250px; /* short hand only works here not at the class level ?!*/

  }
}

@media (min-width: 600px) {
  .listing {
    /*max-width: 600px;*/
    /*min-width: 100px;*/
    width: 1000px;
    display: flex;
    /*flex: 1 1 200px;*/
    flex-wrap: wrap;
    /*border: 4px solid white;*/
    flex-direction: row;
    /*flex:1 0 250px;*/
  }

  section > div {
    /*text-align: center;*/
    flex:0 0 300px; /* short hand only works here not at the class level ?!*/

  }
}