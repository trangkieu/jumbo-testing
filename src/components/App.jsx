import React, {Component} from 'react';
import {BrowserRouter, Switch, Route} from 'react-router-dom';

import './App.css';
import {PopularMovies} from './movie/PopularMovies';
import {MovieDetails} from './movie/MovieDetails';
import {HOME, MOVIE_DETAILS, MOVIE_SEARCH_RESULTS} from './common/Constants';
import SearchResults from "./movie/SearchResults";


export default class App extends Component {
    constructor(props) {
        super(props);
        this.state = {
            popularMovies: null,
            selectedItem: null,
            searchResults: null
        }
    }

    componentDidMount() {

        fetch('https://api.themoviedb.org/3/movie/popular?api_key=6ed12e064b90ae1290fa326ce9e790ff&language=en-US')
            .then(response => response.json())
            .then(data => {
                // let movieResults = data.results.sort((a,b) => (a.popularity < b.popularity) ? 1 : ((b.popularity < a.popularity) ? -1 : 0));
                let movieResults = data.results; // it's already sort in popularity

                this.setState({popularMovies: movieResults});

            }).catch(error => {
            console.error("error while fetching popular movies", error);

        });
    }

    setSelectedDetails(movieDetails) {
        this.setState({
            selectedItem: movieDetails
        });
    }

    setSearchResults(searchResults) {
        this.setState({
            searchResults
        })
    }


    render() {

        return (
            <BrowserRouter>
                <div>
                    <div className="container">
                        <div className="green-deco1">test 1</div>
                        <div className="logo float-right">
                            <img className="logo-image" src="images/green-logo.svg" alt="logo"/>
                        </div>
                        <div className="green-deco2">test 2</div>
                    </div>


                    <div className="container">
                        <Switch>

                            <Route exact path={HOME}
                                   render={() => <PopularMovies popularMovies={this.state.popularMovies}
                                                                detailsHook={this.setSelectedDetails.bind(this)}
                                                                resultsHook={this.setSearchResults.bind(this)}/>}
                            />

                            <Route path={MOVIE_DETAILS}
                                   render={() => <MovieDetails details={this.state.selectedItem}/>}
                            />


                            <Route path={MOVIE_SEARCH_RESULTS}
                                   render={() => <SearchResults results={this.state.searchResults}
                                                                detailsHook={this.setSelectedDetails.bind(this)}/>}
                            />
                        </Switch>
                    </div>


                </div>

            </BrowserRouter>

        );
    }
}

