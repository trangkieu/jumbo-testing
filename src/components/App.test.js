import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
// test file
import { shallow, mount, render } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

import mockPopularMovies from "../data/popularMovies";
import mockMovieDetails from "../data/movieDetails";
import mockSearchMovies from "../data/search-movies";

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App/>, div);
});

it("integration test ", () => {
    const mockFetchPromise = Promise.resolve({ // 3
        json: () => mockPopularMovies,
    });

    global.fetch = jest.fn().mockImplementation(() => mockFetchPromise);

    const wrapper = mount(<App />); // 5

    expect(global.fetch).toHaveBeenCalledTimes(1);
    expect(global.fetch).toHaveBeenCalledWith('https://api.themoviedb.org/3/movie/popular?api_key=6ed12e064b90ae1290fa326ce9e790ff&language=en-US');

    process.nextTick(() => { // 6
        expect(wrapper.state().popularMovies).toEqual(mockPopularMovies.results);
        global.fetch.mockClear(); // 7


// test select details
        it('select first movie ', () => {
            const mockFetchMovieDetails = Promise.resolve({ // 3
                json: () => mockMovieDetails,
            });

            global.fetch = jest.fn().mockImplementation(() => mockFetchMovieDetails);

            const eachMovie = wrapper.find('#eachMovie');
            expect(eachMovie.length).toBe(4); // it finds it alright

            eachMovie[0].simulate('click');

            process.nextTick(() => { // 6
                expect(wrapper.state().selectedItem).toEqual(mockFetchMovieDetails);
                global.fetch.mockClear(); // 7

            });
        })

        // test search for movies
        it('search for a movie using the button', () => {
            const mockFetchSearchMovies = Promise.resolve({ // 3
                json: () => mockSearchMovies,
            });

            global.fetch = jest.fn().mockImplementation(() => mockFetchSearchMovies);

            const searchButton = wrapper.find('#searchImg');
            expect(searchButton.length).toBe(0); // it finds it alright

            searchButton[0].simulate('click');

            process.nextTick(() => { // 6
                expect(wrapper.state().searchResults).toEqual(mockFetchSearchMovies);
                global.fetch.mockClear(); // 7

            });
        })

    });


});


