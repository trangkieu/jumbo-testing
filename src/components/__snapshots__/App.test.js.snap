exports[`Movie details snapshot renders 1`] = `
<div>
  <div
    className="row">
    <div
      className="row__divide-2 about-additional-data">
      <div
        className="card-image flow-text layer-20 bottom">
        <img
          alt="backdrop"
          id="backdropImg"
          src="http://image.tmdb.org/t/p/original/93xA62uLd5CwMOAs37eQ7vPc1iV.jpg" />
      </div>
      <div
        className="card-image flow-text layer-10">
        <img
          alt="poster"
          id="posterImg"
          src="http://image.tmdb.org/t/p/original/lHu1wtNaczFPGFDTrjCSzeLPTKN.jpg" />
      </div>
      <div
        className="card-image flow-text layer-20 bottom">
        uu
        <div
          id="yearRelease">
          2018
        </div>
        <div
          id="votePercentage">
          80
          % User Score
        </div>
        <div
          id="runtime">
          2
          hr 
          15
          min
        </div>
      </div>
      <div
        className="row__divide-2__col-1 layer-30"
        id="movieDescriptions">
        <div>
          <h2>
            Bohemian Rhapsody
          </h2>
          <p>
            Singer Freddie Mercury, guitarist Brian May, drummer Roger Taylor and bass guitarist John Deacon take the music world by storm when they form the rock 'n' roll band Queen in 1970. Hit songs become instant classics. When Mercury's increasingly wild lifestyle starts to spiral out of control, Queen soon faces its greatest challenge yet â€“ finding a way to keep the band together amid the success and excess.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
